solr/modules/opentelemetry/src/java/org/apache/solr/opentelemetry/OtelTracerConfigurator.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.opentelemetry;
+
+import io.opentelemetry.opentracingshim.OpenTracingShim;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
+import io.opentracing.Tracer;
+import java.lang.invoke.MethodHandles;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.apache.solr.core.TracerConfigurator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * OpenTracing TracerConfigurator implementation which exports spans to OpenTelemetry in OTLP
+ * format. This impl re-uses the existing OpenTracing instrumentation through a shim, and takes care
+ * of properly closing the backing Tracer when Solr shuts down.
+ */
+public class OtelTracerConfigurator extends TracerConfigurator {
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  // Copy of environment. Can be overridden by tests
+  Map<String, String> currentEnv = System.getenv();
+
+  @Override
+  public Tracer getTracer() {
+    setDefaultIfNotConfigured("OTEL_SERVICE_NAME", "solr");
+    setDefaultIfNotConfigured("OTEL_TRACES_EXPORTER", "otlp");
+    setDefaultIfNotConfigured("OTEL_EXPORTER_OTLP_PROTOCOL", "grpc");
+    setDefaultIfNotConfigured("OTEL_TRACES_SAMPLER", "parentbased_always_on");
+
+    final String currentConfig =
+        String.join(
+            "; ",
+            getCurrentOtelConfig().entrySet().stream()
+                .map(e -> e.getKey() + "=" + e.getValue())
+                .collect(Collectors.toSet()));
+
+    log.info("OpenTelemetry tracer enabled with configuration: {}", currentConfig);
```

Comment:
Do you think outputting the config warrants INFO level?  ...  I have no doubt it was useful during development but we should consider our log levels when we ship.

Reply:
In fact I do. it's a one-liner and will spell out in cleartext whether and how tracer was configured. Imagine someone asking on user list that they cannot get tracing to work, they have configured an otel collector on port blabla but no traces show up. Now we can ask them to verify this line in the log, and perhaps they discover some ENV variable that is wrong etc.